addons/resource/models/resource_mixin.py

```diff
@@ -43,6 +43,14 @@ def create(self, values):
             values['resource_id'] = resource.id
         return super(ResourceMixin, self).create(values)
 
+    def write(self, vals):
+        if vals.get('tz') is False:
+            raise exceptions.ValidationError(
+                _('The timezone cannot be unset for model: %s (%s), ids: %s')
+                % (self._description, self._name, self.ids)
```

Comment:
this error is rather technical, `self._description` is untranslated (use the `ir.model` instead), also, maybe the rec_name of the record instead of the ids